test/lib/eslint/EslintTestDataGenerator.ts

```diff
@@ -0,0 +1,84 @@
+import { Rule, RuleGroup, RuleTarget, ESRule, ESResult, ESMessage, ESReport } from '../../../src/types';
+
+const engineName = 'TestHarnessEngine'; // TODO: crude code. revisit
+
```

Comment:
Moved existing data creation functions from BaseEslintEngine.test.ts to here for reusablity.